phoenix-core-client/src/main/java/org/apache/phoenix/expression/RowValueConstructorExpression.java

```diff
@@ -259,7 +276,23 @@ public boolean evaluate(Tuple tuple, ImmutableBytesWritable ptr) {
             throw new RuntimeException(e); //Impossible.
         }
     }
-    
+
+    private boolean hasSeparatorBytes(byte[] outputBytes, int outputSize, int k) {
+        if (getChildren().get(k).getDataType() != PVarbinaryEncoded.INSTANCE) {
+            return outputBytes[outputSize - 1] == SchemaUtil.getSeparatorByte(true, false,
+                getChildren().get(k));
+        } else {
+            byte[] sepBytes = SchemaUtil.getSeparatorBytesForVarBinaryEncoded(true, false,
+                getChildren().get(k).getSortOrder());
+            return outputSize >= 2 && outputBytes[outputSize - 1] == sepBytes[1]
+                && outputBytes[outputSize - 2] == sepBytes[0];
+        }
+    }
+
+    private static int getSeparatorBytesLen(Expression expression) {
```

Comment:
you can keep it getSeparatorBytesLength?

Reply:
Sure, will do